src/object/obj_rpc.c

```diff
@@ -99,35 +99,35 @@ crt_proc_struct_obj_io_desc(crt_proc_t proc, struct obj_io_desc *oiod)
 	int		rc;
 
 	rc = crt_proc_uint16_t(proc, &oiod->oiod_nr);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 	rc = crt_proc_uint16_t(proc, &oiod->oiod_tgt_idx);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 	rc = crt_proc_uint32_t(proc, &oiod->oiod_flags);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 
 	rc = crt_proc_get_op(proc, &proc_op);
-	if (rc != 0)
-		return -DER_HG;
-	if (proc_op == CRT_PROC_DECODE && oiod->oiod_nr > 0) {
+	if (unlikely(rc))
+		return rc;
+	if (DECODING(proc_op) && oiod->oiod_nr > 0) {
```

Comment:
This is nice, are the macros for it already in place?

Reply:
Yes, It's already defined but rarely used.